src/upgrade.c

```diff
@@ -35,6 +37,10 @@ static int option_parse_applet(void *ctx, struct apk_db_options *dbopts, int opt
 	case 0x10001:
 		uctx->self_upgrade_only = 1;
 		break;
+	case 0x10002:
+	    	uctx->i_pkg[uctx->ignore] = optarg;
```

Comment:
If the limited size array is kept, I will add some sanity checks here to avoid an overflow.